### ISSUE TITLE
Sticky targeting does not work for rollouts

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/experimenter#7863]
**[Affected Versions]:**

* Firefox Nightly 105.a1 (Build ID: 20220821185924)

* Fennec built from main 6d23c57

* Firefox for Android 105.0a1 (Build #2015899595), 77b959ec9+ AC: 105.0.20220821161547, 2706649f84, GV: 105.0a1-20220821091424 AS: 94.1.0

**[Affected Platforms]:**

* Windows 10

* macOS 11.6.6

* Linux Mint 20.2

* iOS 15.2 - iPhone 13 (Xcode Simulator)

* Android 9 (Samsung Galaxy S8)

**[Prerequisites]:**

* Have a rollout launched to Stage Preview with sticky targeting set.

* Have the browser set to the Stage Preview environment (using Remote Settings Devtools and the “messaging-system.rsexperimentloader.collection_id” set to “nimbus-preview”).
* Be enrolled in the rollout.

**[Steps to reproduce]:**

1. Have the Firefox browser from prerequisites open.
1. Change one of the sticky targeting criteria values to be invalid.
1. Restart the browser.
1. Observe the “about:support” page.
1. Observe the “about:telemetry” page.

**[Expected result]:**

* Step 4: The rollout is still displayed in the “Remote Features” section.

* Step 5: No unenrollment event is sent.

**[Actual result]:**

* Step 4: The rollout is not displayed in the “Remote Features” section.

* Step 5: An unenrollment event is sent for targeting mismatch.

**[Notes]:**

* Based on the functionality, this issue seems reproducible for mobile rollouts as well, but I have not been able to see events for this.

* Attached is a recording of the issue.

!mismatched targeting sticky.gif|width=1574,height=861!

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/EXP-2723)
┆Attachments: <a href="https://mozilla-hub.atlassian.net/rest/api/2/attachment/content/27710">mismatched targeting sticky.gif</a>

**Comment by data-sync-user** [comment]
➤ Tarik Eshaq commented:

cc Yashika Khurana and maybe Barret Rennie  for the desktop side of this, Charlie Humphreys and I briefly took a look at mobile, and currently, all targeting parameters in mobile already have sticky targeting as a part of the parameter itself

**Comment by data-sync-user** [comment]
➤ Benjamin Forehand Jr. commented:

Yashika Khurana Charlie Humphreys did you get a chance to look at this?

**Comment by data-sync-user** [comment]
➤ Yashika Khurana commented:

Thanks, Benjamin Forehand Jr. for checking in, I took a quick look and am not very sure of the cause, let me add Jared Lockhart for his thoughts on this

**Comment by data-sync-user** [comment]
➤ Barret Rennie commented:

This won’t be a problem with desktop – this is an issue with how Experimenter generates the targeting expression.

**Comment by data-sync-user** [comment]
➤ Barret Rennie commented:

Desktop uses activeRollouts and activeExperiments so Experimenter should generate an expression of slug in activeRollouts for sticky targeting in rollouts

**Comment by data-sync-user** [comment]
➤ Daniel Mosedale commented:

This is something we’d like to use with the Import Bookmarks messaging system rollout in 107 release ( https://experimenter.services.mozilla.com/nimbus/import-infrequent-rollout-make-yourself-at-home ), because our targetting is based on people being infrequent users, and if they user it a little more for whatever reason, we don’t want them being unenrolled.

**Comment by jaredlockhart** [comment]
@brennie Oh yes that makes sense thank you.  Then we need some equivalent for mobile.

@tarikeshaq @jeddai What does it mean that each targeting parameter on mobile has sticky targeting as part of the parameter itself?

**Comment by jeddai** [comment]
Reading back through old messages, I believe what we were referring to is that in Experimenter all of the available advanced targeting options for Firefox iOS/Android require sticky enrollment to be enabled.

Mobile has a targeting attribute for `is_already_enrolled`; it looks like that's what is generated by experimenter right now.

https://github.com/mozilla/application-services/blob/main/components/nimbus/src/enrollment.rs#L864-L871

**Comment by data-sync-user** [comment]
➤ Maria Heres commented:

I have verified that the issue is no longer reproducible. Sticky targeting is now functional on desktop. Users are not enrolled if the targeting criteria are not met, but do not get unenrolled if they are no longer valid after users are already enrolled. Tested using Firefox Nightly 108.0a1 (Build ID: 20221026034746) and Firefox Release 106.0.1 (Build ID: 20221019185550) on Windows 10, macOS 11.6.6, Linux 20.2.
